marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx

```diff
@@ -365,4 +364,130 @@ describe("Concept classes in Modeling screen", () => {
+    });
+
+    //relationship modal should open with proper source and target nodes in place
+    relationshipModal.verifySourceEntity("Product").should("be.visible");
+    relationshipModal.verifyTargetNode("testDeleteStyle").should("be.visible");
+
+    //add relationship properties and save
+    relationshipModal.editRelationshipName("quarter");
+
+    //open Optional line to edit foreign key field
+    relationshipModal.toggleOptional();
+    relationshipModal.verifyVisibleOptionalBlock();
+    cy.log("****");
+    relationshipModal.getSourcePropertySelectWrapper().click();
+    relationshipModal.sourceProperty("None").should("not.exist");
+    relationshipModal.sourceProperty(".").should("be.visible");
+
+    relationshipModal.sourceProperty("category").click();
+    relationshipModal.addRelationshipSubmit();
+
+    cy.waitUntil(() => toolbar.getModelToolbarIcon()).click();
+    modelPage.scrollPageBottom();
+    cy.wait(6000);
+
+    graphVis.getPositionsOfNodes().then((nodePositions: any) => {
+      let testDeleteStyleCoordinates: any = nodePositions["testDeleteStyle"];
+      cy.wait(150);
+      cy.waitUntil(() => graphVis.getGraphVisCanvas().click(testDeleteStyleCoordinates.x, testDeleteStyleCoordinates.y, {force: true}));
+    });
+    cy.wait(1500);
+
+    graphViewSidePanel.getDeleteIcon("testDeleteStyle").scrollIntoView().click({force: true});
+    confirmationModal.getDeleteConceptClassWithRelationshipsText().should("exist");
+    confirmationModal.getCloseButton(ConfirmationType.DeleteConceptClassWithRelatedEntityTypes).click();
+    graphViewSidePanel.getSelectedConceptClassHeading("testDeleteStyle").should("exist");
+
+    graphViewSidePanel.closeSidePanel();
+    graphViewSidePanel.getSelectedConceptClassHeading("testDeleteStyle").should("not.exist");
+
+    modelPage.scrollPageBottom();
+    cy.wait(6000);
+
+    //reopen modal to delete
+    graphVis.getPositionOfEdgeBetween("Product,testDeleteStyle").then((edgePosition: any) => {
+      cy.wait(150);
+      graphVis.getGraphVisCanvas().click(edgePosition.x, edgePosition.y, {force: true});
+    });
+
+    cy.log("Deleting relationship between Product and testDeleteStyle");
+    relationshipModal.getDeleteRelationshipIcon().click();
+    confirmationModal.getDeletePropertyWarnText().should("exist");
+    relationshipModal.confirmDeleteRel();
+    confirmationModal.getDeletePropertyWarnText().should("not.exist");
+
+    cy.wait(3000);
+
+    cy.waitForAsyncRequest();
+
+    graphVis.getPositionsOfNodes().then((nodePositions: any) => {
+      let testDeleteStyleCoordinates: any = nodePositions["testDeleteStyle"];
+      cy.wait(150);
+      cy.waitUntil(() => graphVis.getGraphVisCanvas().click(testDeleteStyleCoordinates.x, testDeleteStyleCoordinates.y, {force: true}));
+    });
+    cy.wait(1500);
+
+    graphViewSidePanel.getDeleteIcon("testDeleteStyle").scrollIntoView().click({force: true});
```

Comment:
Continuing with the suggestion: This part, deleting the concept, would be handled by the last it so we wouldn't need it.